aioircd/states.py

```diff
@@ -0,0 +1,331 @@
+                await self.user.send(ErrNoSuchChannel.format(channel))
+                continue
+
+            # Find or create the channel, add the user in it
+            chan = servlocal.channels.get(channel)
+            if not chan:
+                chan = Channel(channel)
+                servlocal.channels[chan.name] = chan
+            chan.users.add(self.user)
+            self.user.channels.add(chan)
+
+            # Send JOIN response to all
+            await chan.send(f":{self.user.nick} JOIN {channel}")
+
+            # Send NAMES list to joiner
+            await self.NAMES(channel)
+
+    @command
+    async def PART(self, channels, reason=None):
+        servlocal = aioircd.servlocal.get()
+
+        for channel in channels.split(','):
+            chan = servlocal.channels.get(channel)
+            if not chan:
+                await self.user.send(ErrNoSuchChannel.format(channel))
+                continue
+
+            if self.user not in chan.users:
+                await self.user.send(ErrNotOnChannel.format(channel))
+                continue
+
+            self.user.channels.remove(chan)
+            chan.users.remove(self.user)
+            if not chan.users:
+                servlocal.channels.pop(chan.name)
+
+            if reason:
+                await chan.send(f":{self.user.nick} PART {channel} :{reason}")
+            else:
+                await chan.send(f":{self.user.nick} PART {channel}")
+
+    @command
+    async def NAMES(self, channel):
+        servlocal = aioircd.servlocal.get()
+        chan = servlocal.channels.get(channel)
+        host = servlocal.host
+        nick = self.user.nick
+
+        if chan:
+            await self.user.send([
+                f":{host} 353 {nick} = {chan} :{some_users}"
+                # merged all users in a str and split it by MAXLINELEN chunks
+                for some_users in textwrap.wrap(
+                    ' '.join(sorted(user.nick for user in chan.users)),
+                    width=aioircd.MAXLINELEN - len(host) - len(nick) - len(channel) - 13
+                )
+            ])
+
+        await self.user.send(f":{host} 366 {nick} {chan} :End of /NAMES list.")
+            
+    @command
+    async def LIST(self):
+        servlocal = aioircd.servlocal.get()
+        host = servlocal.host
+        nick = self.user.nick
+
+        await self.user.send([
+            f":{host} 321 {nick} Channel :Users Name",
+            *[
+                f":{host} 322 {nick} {chan} {len(chan.users)} :"
+                for chan in servlocal.channels.values()
+            ],
+            f":{host} 323 {nick} :End of /LIST"
+        ])
+
+    @command
+    async def PRIVMSG(self, targets, text):
+        servlocal = aioircd.servlocal.get()
+
+        for target in targets.split(','):
+            chan_or_user = (
+                servlocal.channels.get(target)
+                if target.startswith('&') or target.startswith('#') else
```

Comment:
`if target.startswith(('&', '#')) else`
ou stocker les channel prefix dans une liste

Reply:
J'aurais plutôt fait `if target[0] in '&#:`

Reply:
^ pourquoi ?

Reply:
c'est quand même plus élégant d'utiliser une fonction intégrée à python qui est faite pour ça imo

Reply:
Je trouve  ça moche le tuple à l'intérieur, puis bon, pour un caractère je pense pas que ce soit si nécessaire d'utiliser ça